### ISSUE TITLE
Distance visualisation

### ISSUE THREAD

**Issue by DorianBattivelli** [sgoldenlab/simba#299]
Hello, 

I'm using SimBA the last released version of SimBA (just upgraded) on Windows 10, from ma-DLC data.

I have some doubts about the accuracy of the quantification of distances.

1) Is there a way to visualise live the quantification of distance? When using thee "visualize ROI" function, I do not see the distance, but only thee different ROIs. 

2) For now I'm using the smoothing function to reeducxe noise/signal ratio. Do you suggest other tools / strategy? 

Thank you

**Comment by sronilsson** [comment]
Hi @DorianBattivelli!

(i) there is a tool for visualizing ROI features described [HERE](https://github.com/sgoldenlab/simba/blob/master/docs/ROI_tutorial.md#part-5-visualizing-roi-features) which should include distances to ROI centers. It should show, for each frame, how far the animals are from the center of each ROI. Let me know how it runs on your end. 

(ii) It kind of depends on the noise. The very best way (but also the most time-consuming) is to improve the tracking model, and annotate frames where the model gets the body-parts wrong.

**Comment by DorianBattivelli** [comment]
@sronilsson thank you for the quick answer. I'm now trying the ROI feature option. I keep you update about the outcome. 

Other point: I'm facing issue when trying to run "visualize distance" as reported here: 

![Screenshot 2023-10-20 alle 19 17 13](https://github.com/sgoldenlab/simba/assets/66886884/ee7a4ee4-89e0-450b-8647-fead97c2a527)

Can you identify the error?

**Comment by sronilsson** [comment]
Yep thanks for reporting this error @DorianBattivelli I appreciate it - I will see if I can replicate.

**Comment by DorianBattivelli** [comment]
Ok, I fixed the last bug: I had to select a resolution. 
Still I'm facing a problem when selecting only "create last frame" SimBA prompt returns that corresponding plot is saved in "line plot" folder, but no file is there. Saving work only on "create frames" and "video" option. Is that expected?

**Comment by sronilsson** [comment]
@DorianBattivelli -  which version of SimBA are you running? There shouldn't be a "As input" option and I can't see it in the latest version.

**Comment by sronilsson** [comment]
No - `last frame` should create a single image called something like `MyVideoName_final_img.png` located in `/project_folder/frames/output/line_plot`.

**Comment by sronilsson** [comment]
Ah hang on, I could replicate the error when the directory `/project_folder/frames/output/line_plot` doesn't exist and trying to write the file there, let me see if I can insert a fix.

**Comment by DorianBattivelli** [comment]
I'm using SimBA 1.75.3 on Windows 10

**Comment by sronilsson** [comment]
... or it turns out the file isn't saved when "multi-processed" is ticked.

**Comment by DorianBattivelli** [comment]
Yes in my case I ticked "multi-processed"

**Comment by sronilsson** [comment]
Yes, it turns out I forgot to pass `save_img=True`, when multiprocessing. It only stored the final image in memory..


How does it look in version 1.75.4? Thanks again.

**Comment by DorianBattivelli** [comment]
It works! No "as input" in the list, and file saved with "create last frame" even with multi-processed ticked, thank you :)

**Comment by sronilsson** [comment]
Tha is for letting me know! Also let me know how the ROI features runs or if there are any issues.

**Comment by DorianBattivelli** [comment]
O, I realized that the issue for saving persists when I try to generate the last frame for "all the videos" instead of once at a time

**Comment by sronilsson** [comment]
Ah, thank you! Let's fix that.

**Comment by sronilsson** [comment]
It should be fixed in 1.75.5 but please let me know, and please let me know if you see any other bugs.

**Comment by DorianBattivelli** [comment]
Thanks for the fix, I'll try it soon. Meantime, another error/issue I face:
I tried to visualize features for one video, but the prompt returns:

<img width="554" alt="Screenshot 2023-10-20 alle 21 55 04" src="https://github.com/sgoldenlab/simba/assets/66886884/89da241d-10b2-45b2-afd7-7fe34bd67fd0">

And indeed, on this specific video, Up-HT is not present, but it is for other videos. Is that mandatory to have exactly the same ROIs for all videos of this project to be able to run this command?

**Comment by sronilsson** [comment]
It appears so! I could remove the requirement. But - the reason it is required is that the function mainly targets people who want to create features from ROIs for building classifiers. In this scenario, it is not possible to create some ROI-based features for some videos, and skip them for other videos.

**Comment by DorianBattivelli** [comment]
Yes it would be nice ;) thank you

**Comment by DorianBattivelli** [comment]
Also I was wondering, when ploting paths, would that be possible to set an option allowing to choose a frame from the video as background instead of a plain color?

**Comment by sronilsson** [comment]
Ah thats a good point. Yes I can insert that. I have to do the ROI feature plotting first. It will take a little longer because I realized you can't choose which body-parts to use in the pop-up menu, it says you have to analyze body-parts first which is not particularly good...

**Comment by DorianBattivelli** [comment]
Not sure to understand, in the path-plot menu, I can choose the BP to plote

<img width="319" alt="Screenshot 2023-10-20 alle 22 47 26" src="https://github.com/sgoldenlab/simba/assets/66886884/7291c9a3-cdb3-443c-8f0e-96abbcfa7782">

**Comment by DorianBattivelli** [comment]
Ah I see, instead I suggested frame background for the paths visualisation plots, would this be possible?

**Comment by sronilsson** [comment]
yes, that's possible we should include it, I let you know when done, but I can't get to it today.

**Comment by DorianBattivelli** [comment]
Amazing thank you !

**Comment by sronilsson** [comment]
@DorianBattivelli - before I go on, is this the sort of thing you are looking for:

You can select to use `video` as background. If you select `video`, you also get the option to set the opacity of the background (set it to a lower value if you want to make it a little less salient). 

<img width="489" alt="image" src="https://github.com/sgoldenlab/simba/assets/34761092/ab410bac-8187-44e6-81fb-ccdb175874d2">

Right now it takes the first frame of the video as default, do you need an option to take a different frame? 

The output looks something like this (I have a troubleshooting project with your name, but I have most likely I mixed up which tracking files belong to which videos).

![HybCD1-B2-D6-Urine_final_frame](https://github.com/sgoldenlab/simba/assets/34761092/7cdd9e6f-17d5-461b-980b-016a24b6dcf2)

**Comment by sronilsson** [comment]
... or is that you want the entire video (updating frames) as background ?

**Comment by DorianBattivelli** [comment]
> @DorianBattivelli - before I go on, is this the sort of thing you are looking for:
> 
> 
> 
> You can select to use `video` as background. If you select `video`, you also get the option to set the opacity of the background (set it to a lower value if you want to make it a little less salient). 
> 
> 
> 
> <img width="489" alt="image" src="https://github.com/sgoldenlab/simba/assets/34761092/ab410bac-8187-44e6-81fb-ccdb175874d2">
> 
> 
> 
> Right now it takes the first frame of the video as default, do you need an option to take a different frame? 
> 
> 
> 
> The output looks something like this (I have a troubleshooting project with your name, but I have most likely I mixed up which tracking files belong to which videos).
> 
> 
> 
> ![HybCD1-B2-D6-Urine_final_frame](https://github.com/sgoldenlab/simba/assets/34761092/7cdd9e6f-17d5-461b-980b-016a24b6dcf2)
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
This is the solution I need, if it's also possible to select a precise frame it's perfect

**Comment by DorianBattivelli** [comment]
And by any chance if it's also possible to specify a precise colore for traces from rgb palette it becomes paradise

**Comment by sronilsson** [comment]
Haha sounds good - something like this also, where you fill in the rgb values if set to `Custom`  and option to set the exact frame index? Hang on, have to write some unit tests through... so users can't insert color `crazy number,1999, -123` etc. 

<img width="531" alt="image" src="https://github.com/sgoldenlab/simba/assets/34761092/486e433b-55d8-4923-9970-4f092c9aa5bb">

**Comment by sronilsson** [comment]
I included it in the latest version. Let me know if you get any errors or if something is unclear though.
